integration-tests/src/test/java/oracle/weblogic/kubernetes/ItHorizontalPodAutoscaler.java

```diff
@@ -0,0 +1,259 @@
+    logger.info("Create encryption secret");
+    encryptionSecretName = "encryptionsecret";
+    createSecretWithUsernamePassword(encryptionSecretName, domainNamespace,
+        "weblogicenc", "weblogicenc");
+
+    domain = createDomainResource(
+        domainUid,
+        domainNamespace,
+        MII_BASIC_IMAGE_NAME + ":" + MII_BASIC_IMAGE_TAG,
+        adminSecretName,
+        new String[]{TEST_IMAGES_REPO_SECRET_NAME},
+        encryptionSecretName
+    );
+
+    // create cluster resouce with limits and requests in serverPod
+    ClusterResource clusterResource =
+        createClusterResource(clusterResName, wlClusterName, domainNamespace, replicaCount);
+    clusterResource.getSpec()
+        .serverPod(new ServerPod().resources(
+            new V1ResourceRequirements()
+                .putLimitsItem("cpu", Quantity.fromString("2"))
+                .putLimitsItem("memory", Quantity.fromString("2Gi"))
+                .putRequestsItem("cpu", Quantity.fromString("250m"))
+                .putRequestsItem("memory", Quantity.fromString("768Mi"))));
+    logger.info("Creating cluster {0} in namespace {1}", clusterResName, domainNamespace);
+    createClusterAndVerify(clusterResource);
+
+    // set cluster references
+    domain.getSpec().withCluster(new V1LocalObjectReference().name(clusterResName));
+
+    // verify the domain custom resource is created
+    createDomainAndVerify(domain, domainNamespace);
+
+    // check admin server is up and running for domain1
+    checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace);
+
+    // check managed servers are up and running for domain1
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodReadyAndServiceExists(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+  }
+
+
+  /**
+   * Test autoscaling using HPA by increasing the CPU usage.
+   */
+  @Test
+  void testHPAWithMetricsServer() {
+    // install metrics server
+    installMetricsServer();
+
+    // create hpa with autoscale
+    createHPA();
+
+    // create load on cpu and verify scale up/scale down
+    createLoadOnCpuAndVerifyAutoscaling();
+  }
+
+  /**
+   * Delete metrics server.
+   */
+  @AfterAll
+  public static void cleanUp() {
+    if (!SKIP_CLEANUP) {
+      getLogger().info("After All cleanUp() method called");
+      // delete metrics server
+      CommandParams params = new CommandParams().defaults();
+      params.command("kubectl delete -f " + METRICS_SERVER_YAML);
+      ExecResult result = Command.withParams(params).executeAndReturnResult();
+      if (result.exitValue() != 0) {
+        getLogger().info(
+            "Failed to uninstall metrics server, result " + result);
+      } else {
+        getLogger().info("uninstalled metrics server");
+      }
+    }
+  }
+
+  private void installMetricsServer() {
+    // install metrics server
+    CommandParams params = new CommandParams().defaults();
+    params.command("kubectl apply -f " + METRICS_SERVER_YAML);
```

Comment:
Can u put retry logic, in case we hit the network latency.

Reply:
so far have not seen any issues with this, will see if we really need